pkg/storage/storage.go

```diff
@@ -317,11 +318,26 @@ type ChangelogBackend interface {
 	// if no changes are found, it should return storage.ErrNotFound and an empty continuation token.
 	// It the objectType and the type in the continuation token don't match, it should return ErrMismatchObjectType.
 	ReadChanges(ctx context.Context, store string, filter ReadChangesFilter, options ReadChangesOptions) ([]*openfgav1.TupleChange, []byte, error)
+}
 
+type ReadChangesTokenSerializer interface {
 	// SerializeReadChangesContToken serializes the continuation token into a format readable by ReadChanges
 	SerializeReadChangesContToken(ulid string, objType string) ([]byte, error)
 }
```

Comment:
My suggestion will go against the interface segregation principle, but: why is this a new interface and not part of the ChangelogBackend interface? Since this is seems to be tightly coupled to it - i.e. you shouldn't be able to write a ChangelogBackend implementation _without a token serializer_?